### PR TITLE
feat(frontend): Trigonometric functions in degrees

### PR DESCRIPTION
**Pull request by CAJan93**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

Implement trigonometric functions in degree. Merge after original one https://github.com/risingwavelabs/risingwave/pull/8838 

https://github.com/risingwavelabs/risingwave/issues/8806 

- `sind` / `cosd` / `tand` / `cotd` / `asind` / `acosd` / `atand` / `atan2d`

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [x] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

Documents are here: https://github.com/risingwavelabs/risingwave/issues/9678

### CODE REVIEW

File: src/expr/src/vector_op/trigonometric.rs [hunk 1]
```diff
@@ -56,9 +56,55 @@ pub fn atan2_f64(input_x: F64, input_y: F64) -> F64 {
     input_x.0.atan2(input_y.0).into()
 }
 
+#[function("sind(float64) -> float64")]
+pub fn sind_f64(input: F64) -> F64 {
+    f64::sin(f64::to_radians(input.0)).into()
+}
```
Comment (xiangjinwu) [inline src/expr/src/vector_op/trigonometric.rs:62]: The PostgreSQL implementation is nontrivial, as it is [intended](https://www.postgresql.org/docs/current/functions-math.html) to be more accurate than conversion first.
> Another way to work with angles measured in degrees is to use the unit transformation functions radians() and degrees() shown earlier. However, using the degree-based trigonometric functions is preferred, as that way avoids round-off error for special cases such as sind(30).

For reference, see [this](https://github.com/postgres/postgres/blob/REL_15_2/src/backend/utils/adt/float.c#L2444).

Comment (CAJan93) [reply src/expr/src/vector_op/trigonometric.rs:62]: Thank you very much for pointing that out. Sorry, I did not have time to continue working on this. I will try to introduce the function one by one whenever I have time for it. My first short at `Sind`: https://github.com/risingwavelabs/risingwave/pull/9056

Comment (CAJan93) [reply src/expr/src/vector_op/trigonometric.rs:62]: `Cosd`: https://github.com/risingwavelabs/risingwave/pull/9064/

**Comment by CAJan93** [comment]
Closing this one. Will break it up in small PRs. Next up is [asind](https://github.com/risingwavelabs/risingwave/pull/9741)

**Pull request closed without merging**